include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -37,16 +38,21 @@ namespace __ranges
 // walk_n
 //------------------------------------------------------------------------
 
-template <typename _ExecutionPolicy, typename _Function, typename... _Ranges>
+template <int N = 0, typename _ExecutionPolicy, typename _Function, typename... _Ranges>
 oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy, void>
 __pattern_walk_n(_ExecutionPolicy&& __exec, _Function __f, _Ranges&&... __rngs)
 {
     auto __n = oneapi::dpl::__ranges::__get_first_range_size(__rngs...);
     if (__n > 0)
-        oneapi::dpl::__par_backend_hetero::__parallel_for(::std::forward<_ExecutionPolicy>(__exec),
+    {
+        using __new_name = oneapi::dpl::__par_backend_hetero::new_kernel_name<_ExecutionPolicy, N>;
+        auto __new_exec =
+            oneapi::dpl::execution::make_device_policy<__new_name>(::std::forward<_ExecutionPolicy>(__exec));
```

Comment:
I think it won't work if `_ExecutionPolicy` is an FPGA policy. In that case we should call `make_fpga_policy`.

Seems, here we need something like `make_wrapped_policy`, but accepting `N`